File: lib/models/labor_entry.dart
class LaborEntry {
  final String id;
  final String companyId;
  final String employeeId;
  final String jobId;
  final String? jobCodeId;
  final DateTime workDate;
  final double hours;
  final double? hourlyRate;
  final String sourceType;
  final String? notes;
  final DateTime createdAt;

  const LaborEntry({
    required this.id,
    required this.companyId,
    required this.employeeId,
    required this.jobId,
    required this.jobCodeId,
    required this.workDate,
    required this.hours,
    required this.hourlyRate,
    required this.sourceType,
    required this.notes,
    required this.createdAt,
  });

  factory LaborEntry.fromMap(Map<String, dynamic> row) {
    DateTime dt(dynamic v) =>
        DateTime.tryParse(v?.toString() ?? '')?.toLocal() ?? DateTime.now();
    return LaborEntry(
      id: row['id']?.toString() ?? '',
      companyId: row['company_id']?.toString() ?? '',
      employeeId: row['employee_id']?.toString() ?? '',
      jobId: row['job_id']?.toString() ?? '',
      jobCodeId: row['job_code_id']?.toString(),
      workDate: dt(row['work_date']),
      hours: (row['hours'] as num?)?.toDouble() ?? 0,
      hourlyRate: (row['hourly_rate'] as num?)?.toDouble(),
      sourceType: row['source_type']?.toString() ?? 'manual',
      notes: row['notes']?.toString(),
      createdAt: dt(row['created_at']),
    );
  }
}

File: KaiFlow.Timesheets.Maui/Models/LaborEntry.cs
using Newtonsoft.Json;
using Supabase.Postgrest.Attributes;
using Supabase.Postgrest.Models;

namespace KaiFlow.Timesheets.Models;

public enum LaborSourceType { Manual, Punch }

[Table("labor_entries")]
public class LaborEntry : BaseModel
{
    [PrimaryKey("id")]
    public Guid Id { get; set; }

    [Column("employee_id")]
    public Guid EmployeeId { get; set; }

    [Column("job_id")]
    public Guid? JobId { get; set; }

    [Column("job_code_id")]
    public Guid? JobCodeId { get; set; }

    [Column("work_date")]
    public DateOnly WorkDate { get; set; }

    [Column("hours")]
    public double Hours { get; set; }

    [Column("hourly_rate")]
    public double HourlyRate { get; set; }

    [Column("source_type")]
    public string SourceTypeRaw { get; set; } = "manual";

    [Column("notes")]
    public string? Notes { get; set; }

    [Column("company_id")]
    public Guid CompanyId { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    public LaborSourceType SourceType => SourceTypeRaw == "punch"
        ? LaborSourceType.Punch
        : LaborSourceType.Manual;

    public double TotalCost => Hours * HourlyRate;
}
